shared/settings/advanced/index.tsx

```diff
@@ -54,43 +56,83 @@ const UseNativeFrame = (props: Props) => {
+  showCertPinningModal: boolean
+}
+
+class Advanced extends React.Component<Props, AdvancedState> {
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      disableCertPinning: undefined,
+      showCertPinningModal: false,
+    }
+  }
+
+  confirmDisableCertPinning = (disableCertPinning: () => void) => {
+    this.setState({disableCertPinning, showCertPinningModal: true})
+  }
+
+  render() {
+    const disabled =
+      this.props.lockdownModeEnabled == null || this.props.hasRandomPW || this.props.settingLockdownMode
+    return (
+      <Kb.Box style={styles.advancedContainer}>
+        {this.state.showCertPinningModal && (
+          <Kb.Box style={{zIndex: 1}}>
+            {' '}
+            <Kb.ConfirmModal
+              confirmText="Yes, allow TLS MITM"
+              description="This means your proxy will be able to view all traffic between you and Keybase servers. It
+          is not recommended to use this option unless absolutely required."
+              header={<Kb.Icon type="iconfont-exclamation" sizeType="Big" color={Styles.globalColors.red} />}
+              onCancel={() => this.setState({showCertPinningModal: false})}
+              onConfirm={() => {
+                this.setState({showCertPinningModal: false})
+                this.state.disableCertPinning()
+              }}
+              prompt="Are you sure you want to allow TLS MITM?"
+            />
+          </Kb.Box>
+        )}
+
+        <Kb.Box style={styles.progressContainer}>
+          {this.props.settingLockdownMode && <Kb.ProgressIndicator />}
+        </Kb.Box>
+        <Kb.Box style={styles.checkboxContainer}>
           <Kb.Checkbox
-            label="Open Keybase on startup"
-            checked={props.openAtLogin}
-            onCheck={props.onSetOpenAtLogin}
+            checked={this.props.hasRandomPW || !!this.props.lockdownModeEnabled}
+            disabled={disabled}
+            label={
+              'Forbid account changes from the website' +
+              (this.props.hasRandomPW ? ' (you need to set a password first)' : '')
+            }
+            onCheck={this.props.onChangeLockdownMode}
+            style={styles.checkbox}
           />
         </Kb.Box>
-      )}
-      <Developer {...props} />
-    </Kb.Box>
-  )
+        {!!this.props.setLockdownModeError && (
+          <Kb.Text type="BodySmall" style={styles.error}>
+            {this.props.setLockdownModeError}
+          </Kb.Text>
+        )}
+        {isLinux && <UseNativeFrame {...this.props} />}
+        {!Styles.isMobile && !isLinux && (
+          <Kb.Box style={styles.openAtLoginCheckboxContainer}>
+            <Kb.Checkbox
+              label="Open Keybase on startup"
+              checked={this.props.openAtLogin}
+              onCheck={this.props.onSetOpenAtLogin}
+            />
+          </Kb.Box>
+        )}
+        <ProxySettings {...{...this.props, confirmDisableCertPinning: this.confirmDisableCertPinning}} />
```

Comment:
shouldn't pass all props but if you need to you can do
```
{...this.props} confirmDisableCertPinning={this.confirmDisableCertPinning}
```

Reply:
Switched to just `{...this.props}` since I got rid of the closure via migrating to a modal route.